aiohttp/client_reqrep.py

```diff
@@ -359,9 +359,11 @@ def update_headers(self, headers: Optional[LooseHeaders]) -> None:
         netloc = self.url.host_subcomponent
         assert netloc is not None
         # See https://github.com/aio-libs/aiohttp/issues/3636.
-        netloc = netloc.rstrip(".")
-        if self.url.port is not None and not self.url.is_default_port():
-            netloc += ":" + str(self.url.port)
+        if netloc[-1] == ".":
+            netloc = netloc.rstrip(".")
```

Comment:
This seems like the kind of micro-optimisation that looks like redundant code and could easily be refactored back to the previous form by a future contributor...

Reply:
Definitely needs a comment here.  I'll also better explain why its done.